src/asm/fixpoint.c

```diff
@@ -21,105 +20,96 @@
 #define M_PI 3.14159265358979323846
 #endif
 
-#define fix2double(i)	((double)((i) / fix_PrecisionFactor()))
-#define double2fix(d)	((int32_t)round((d) * fix_PrecisionFactor()))
+#define fix2double(i)	((double)((i) / pow(2.0, q)))
+#define double2fix(d)	((int32_t)round((d) * pow(2.0, q)))
```

Comment:
```suggestion
#define fix2double(i, q)	((double)((i) / pow(2.0, q)))
#define double2fix(d, q)	((int32_t)round((d) * pow(2.0, q)))
```

I was very confused by where the `q` argument was used. I understand why you did this, but I don't think it's a good idea.